src/hotspot/share/gc/shenandoah/heuristics/shenandoahHeuristics.cpp

```diff
@@ -244,17 +258,20 @@ void ShenandoahHeuristics::log_trigger(const char* fmt, ...) {
 }
 
 void ShenandoahHeuristics::record_success_concurrent() {
+  _start_gc_is_pending = false;
```

Comment:
The name _start_gc_is_pending implies that it should be set false as soon as a gc cycle starts, not when it finishes. Maybe _gc_pending? Or maybe setting it false at the end of a gc cycle is a bug? :)

Reply:
You make a good point.  I'll change the control flow to cancel the trigger as soon as we start up the GC.